fix electrum conftime_req filter for needs_block_height

zoedberg opened the pull request:
<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Description

<!-- Describe the purpose of this PR, what's being adding and/or fixed -->

This PR fixes a bug introduced in https://github.com/bitcoindevkit/bdk/commit/04994e4f7f1073e7bef29528339369994d4b3666#diff-57bf66f87897e694c5ebfdfe0fd366774dda30b43eab93c1a0fdc802d0eb8c8dR171

In that commit `block_times.get(height).is_none()` was converted to `block_times.contains_key(height)`, inverting the code logic.

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

zoedberg committed:
fix electrum conftime_req filter for needs_block_height

ValuedMammal commented:
LGTM

oleonardolima reviewed (commented):
utACK 8d190bacedf80039626397ba87d5b7139cb644e6

notmandatory committed:
ci: update examples/rpcwallet to use electrds/bitcoind_22_1

notmandatory committed:
ci(clippy): fix new stricter needless_lifetime errors for rust 1.83

https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
rust-lang/rust-clippy#13286

notmandatory committed:
ci(clippy): fix updated ptr_arg check for rust 1.83

https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
rust-lang/rust-clippy#13313

notmandatory committed:
ci(clippy): fix missing docs errors for rust 1.83

notmandatory committed:
ci: pin msrv dep version for rustls and hashbrown

notmandatory committed:
ci: pin msrv dep version for ureq

notmandatory committed:
ci(clippy): fix disallow ref to static mut

https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html

notmandatory committed:
ci: downgrade all workflows to ubuntu to 20.04

notmandatory commented:
I had to make some small fixes to get CI to complete, see  4aa8fba7f0005f27bc4d581891535dabd426ac0c...20100aabd54d507e01894993829b2968cee10f25.

notmandatory reviewed (approved):
ACK 20100aabd54d507e01894993829b2968cee10f25

@zoedberg thank you for catching and fixing this! How did you find it?

zoedberg commented:
> thank you for catching and fixing this! How did you find it?

I've just updated [rgb-lib](https://github.com/RGB-Tools/rgb-lib) to bdk 0.30.0 and almost all its tests started failing. Precisely the call `wallet.sync(bdk_blockchain, SyncOptions { progress: None })` was returning a `Generic("electrum server misbehaving")` error. At that point I've inspected bdk's code, checked the points where the error could have been raised, then checked the blame to identify the latest commit that modified those lines and finally inspecting that commit I've found the incorrect change (https://github.com/bitcoindevkit/bdk/commit/04994e4f7f1073e7bef29528339369994d4b3666#diff-57bf66f87897e694c5ebfdfe0fd366774dda30b43eab93c1a0fdc802d0eb8c8dR171).

oleonardolima reviewed (commented):
utACK 20100aabd54d507e01894993829b2968cee10f25

@notmandatory any specific reason to downgrade the runners to `ubuntu-20.04`?

nymius reviewed (commented):
ACK 20100aabd54d507e01894993829b2968cee10f25

> @notmandatory any specific reason to downgrade the runners to ubuntu-20.04?

Curious about this too

notmandatory commented:
@oleonardolima and @nymius I had to downgrade the runner to the ubuntu-20.04 for two issues:
1. the lastest version of python breaks the old pyo3 needed for hwi
2. something with the latest c compiler breaks the old rocksdb used in the old cbf impl 

I don't see any reason to fix these two things to work with the latest ubuntu since they're rarely used.
